test-e2e/app/Tests/SocialNetworks/Facebook.e2e-spec.ts

```diff
@@ -0,0 +1,93 @@
+import { $, browser } from 'protractor';
+
+import { MatrixPage } from '../../Pages';
+import { Footer } from '../../Pages/Components';
+import { disableAnimations, getRandomNumber, waitForVisible } from '../../Helpers';
+import {
+  getRandomString,
+  scrollIntoView,
+  switchWindow,
+  waitForInvisibility,
+  waitTillWindowClosed,
+  waitForURLContain
+} from '../../Helpers/commonHelper';
+import { FacebookPage } from '../../Pages/SocialNetworks/Facebook.page';
+import { HamburgerMenu } from '../../Pages/Components/HamburgerMenu.e2e.component';
+
+let messageText: string;
+let familyNumber: number;
+describe('Facebook test', () => {
+  beforeAll(async () => {
+    messageText = getRandomString(12);
+    familyNumber = getRandomNumber();
+    await browser.get(MatrixPage.url);
+    waitForInvisibility(MatrixPage.spinner);
+    await browser
+      .actions()
+      .mouseMove(MatrixPage.familyLink.get(10))
+      .perform();
+    await disableAnimations();
+  });
+
+  it('Check user can share to facebook from floating footer', async () => {
+    await MatrixPage.familyLink.get(familyNumber).click();
+
+    const urlBefore = await browser.getCurrentUrl();
+    await Footer.facebookIcon.click();
+
+    await switchWindow(1);
+    await browser.waitForAngularEnabled(false);
+    await waitForVisible(FacebookPage.buttonLogIn);
+    await FacebookPage.login();
+    await waitForVisible(FacebookPage.messageBox);
+    await FacebookPage.addMessage(messageText);
+    await FacebookPage.post();
+    await waitTillWindowClosed();
+    await browser.sleep(3000); // Delay needed to get the post to appear on facebook.
+
+    await switchWindow(0);
+    await browser.get(FacebookPage.baseURL);
+
+    await waitForVisible(FacebookPage.postLinkToDollarStreet);
+    await FacebookPage.postLinkToDollarStreet.click();
+    await switchWindow(1);
+    await browser.waitForAngularEnabled(true);
+    await waitForURLContain('matrix');
```

Comment:
looks like you copy\paste about 10 lines of code in 'it's - better to move such logic to PO